src/utlist.h

```diff
@@ -349,6 +349,38 @@ do {
   }                                                                                            \
 } while (0)
 
+#define LL_INSERT_INORDER(head,add,cmp)                                                        \
+  LL_INSERT_INORDER2(head,add,cmp,next)
+
+#define LL_INSERT_INORDER2(head,add,cmp,next)                                                        \
+  do {                                                                                         \
+    LDECLTYPE(head) _pos;                                                                      \
+    _CASTASGN(_pos, head);                                                                     \
+    if(head) {                                                                                 \
+      LL_LOWER_BOUND(_pos,add,cmp);                                                            \
+      LL_APPEND_ELEM(head,_pos,add);                                                           \
+    } else {                                                                                   \
+      (head) = (add);                                                                          \
+      (head)->next = NULL;                                                                     \
+    }                                                                                          \
+  } while (0)
+
+#define LL_LOWER_BOUND(head,node,cmp)                                                          \
+  LL_LOWER_BOUND2(head,node,cmp,next)
+
+#define LL_LOWER_BOUND2(head,node,cmp,next)                                                    \
+  do {                                                                                         \
+    if (cmp(head, node) > 0) {                                                                 \
```

Comment:
Here and throughout: `> 0` should be `>= 0` so that the following test case passes:

```
struct S { int v; struct S *next; };

int scmp(const struct S *a, const struct S *b)
{
    return (a->v < b->v) ? -1 : (a->v > b->v);
}

int main()
{
    struct S a = { 1, NULL };
    struct S b = { 1, NULL };
    struct S c = { 1, NULL };

    struct S *head = NULL;
    LL_INSERT_INORDER(head, &a, scmp);
    LL_INSERT_INORDER(head, &b, scmp);
    LL_INSERT_INORDER(head, &c, scmp);
    assert(head == &c && c.next == &b && b.next == &a && a.next == NULL);
}
```

Inserting 100 copies of "x" into the same empty list should take O(100) time, not O(100*100) time.
Please add a test case for this, too.